src/subcommand/init.rs

```diff
@@ -160,15 +160,11 @@ function {}
         or return $status
 
     else
-        # FIXME: use string-collect from fish 3.1.0 once it has wider adoption
-        set -l IFS ''
         set -l result (zoxide query $argv)
 
-        if test -d $result
+        if begin; test -d $result; and string length -q -- $result; end
```

Comment:
I think `begin` and `end` are unnecessary here:

```fish
if test -d $result; and string length -q -- $result
    _z_cd $result
    or return $status
end
```

Reply:
You're right. I thought I tried it earlier but I guess I did something wrong.